### PR TITLE
Fix impl of is-greater-than-dot-net-5 version check.

### PR DESCRIPTION
**Pull request by rohanp-msft**
See the comment by Daniel Plaisted in: #707.

**How verified**
MRTCore C# lib and app under .NET 5 and .NET 6.

### CODE REVIEW

**Review by axelandrejs — APPROVED** [review]

File: dev/MRTCore/packaging/Microsoft.ApplicationModel.Resources.PriGen.targets [hunk 1]
```diff
@@ -8,9 +8,8 @@
   <PropertyGroup>
     <TargetPlatformIdentifierAdjusted>$(TargetPlatformIdentifier)</TargetPlatformIdentifierAdjusted>
 
-    <!-- Any .NET 5 or greater version is valid here! For example, we wll use UAP if TargetFramework contains net5.0 and net5.1 (or, net50 and net51 if that format is used). -->
-    <TargetFrameworkDotNetString>$([System.Text.RegularExpressions.Regex]::Match("$(TargetFramework)", "net\d"))</TargetFrameworkDotNetString>
-    <TargetPlatformIdentifierAdjusted Condition="'$(TargetFrameworkDotNetString)' != '' AND '$(TargetFrameworkDotNetString.Substring(3))' >= '5'">UAP</TargetPlatformIdentifierAdjusted>
+    <!-- Any .NET 5 or greater version is valid here! -->
+    <TargetPlatformIdentifierAdjusted Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), '5.0'))">UAP</TargetPlatformIdentifierAdjusted>
```
[the remark this answers is not in the document]
Comment (rohanp-msft) [reply dev/MRTCore/packaging/Microsoft.ApplicationModel.Resources.PriGen.targets:12]: those quotes work for numeric comparison based on my testing

[the remark this answers is not in the document]
Comment (rohanp-msft) [reply dev/MRTCore/packaging/Microsoft.ApplicationModel.Resources.PriGen.targets:12]: also that's direct copy paste from .net code

[the remark this answers is not in the document]
Comment (huichen123) [reply dev/MRTCore/packaging/Microsoft.ApplicationModel.Resources.PriGen.targets:12]: but logically that's wrong. string comparison vs. numerical comparison.

[the remark this answers is not in the document]
Comment (huichen123) [reply dev/MRTCore/packaging/Microsoft.ApplicationModel.Resources.PriGen.targets:12]: try set version to 5, and see the result.

[the remark this answers is not in the document]
Comment (rohanp-msft) [reply dev/MRTCore/packaging/Microsoft.ApplicationModel.Resources.PriGen.targets:12]: that works too.
i searched in the vs repo, both are used (string one is used more).
functionally this is fine.
i think the string is just converted to a double before the comparison is done.
i agree that not using the string is the right thing though.
if you want, i can change it to not use quotes.

[the remark this answers is not in the document]
Comment (dsplaisted) [reply dev/MRTCore/packaging/Microsoft.ApplicationModel.Resources.PriGen.targets:12]: MSBuild doesn't have separate data types for strings versus numbers, it depends on the context.  The `VersionGreaterThanOrEquals` method will convert to a version number.

**Pull request merged**